src/legacy/core_plugins/kibana/public/discover/np_ready/angular/get_painless_error.ts

```diff
@@ -18,24 +18,22 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { get } from 'lodash';
 
 export function getPainlessError(error: Error) {
-  const rootCause: Array<{ lang: string; script: string }> | undefined = get(
-    error,
-    'resp.error.root_cause'
-  );
-
-  if (!rootCause) {
+  const errorBody = (error as any).body;
+  if (!errorBody || errorBody.statusCode !== 400) {
     return;
   }
 
-  const [{ lang, script }] = rootCause;
-
-  if (lang !== 'painless') {
+  // The response from the new endpoint does not contain a rootCause.
```

Comment:
Errors returned from the new service are different. Should we adjust them?

Reply:
Yes we should probably adjust them to return errors similarly as before.